src/twister2/yaml_file.py

```diff
@@ -48,12 +46,14 @@ def _generate_test_variants_for_platforms(
         if platform.identifier in twister_config.default_platforms
     ]
 
-    test_name = spec['name']
     for platform in platforms:
+        test_name = spec['name']
         spec['name'] = f'{test_name}[{platform.identifier}]'
         spec['original_name'] = test_name
         spec['platform'] = platform.identifier
+
         yaml_test_spec = YamlTestSpecification(**spec)
+        yaml_test_spec.timeout = math.ceil(yaml_test_spec.timeout * platform.testing.timeout_multiplier)
```

Comment:
Isn't it redundant? Probably this timeout should be set in `SpecificationProcessor` in `process` method, but maybe I missed something...

Reply:
It is only implemented in `SpecificationProcessor` now.